src/main/java/org/jboss/msc/service/ServiceControllerImpl.java

```diff
@@ -1775,11 +1775,13 @@ private void startFailed(StartException e, ServiceName serviceName, StartContext
                 }
                 context.state = ContextState.FAILED;
             }
-            final ArrayList<Runnable> tasks;
             synchronized (ServiceControllerImpl.this) {
-                final boolean leavingRestState = isStableRestState();
                 startException = e;
                 failCount++;
+            }
+            final ArrayList<Runnable> tasks;
+            synchronized (ServiceControllerImpl.this) {
```

Comment:
This change is not wrong.  However, creating two synchronization blocks generally doesn't guarantee that the lock is yielded in between them.  The JIT might even merge them together at run time.  If something needs to happen in between the locks, then some other kind of mechanism has to be used to wait for it.

So the change is not wrong, but it's not really right either. :)